### PR TITLE
Large miner fixes

### PR DESCRIPTION
**Pull request by ProTriforcer**
## What
Fix some issues of large miners

## Outcome

- Large miners no longer have double the intended radius.
- Large miners drain energy properly.
- Chunk mode updates the area properly.
- Radius, chunk and silk touch modes can be changed when the miner is idling for reasons other than a soft hammer.
- Miners are considered not done when the radius or chunk mode is changed.
- Miners pause when they are done, so they don't restart and lock the radius as soon as you change it.
- Slightly finer radius adjustments when not in chunk mode.
- Radius change messages and the GUI show the appropriate radius depending on chunk mode.

image_2022-12-27_125826509

### CODE REVIEW

**Comment by ALongStringOfNumbers** [comment]
Most of the things in this PR look good, however concerning the radius of the large miners, the T1 miner is supposed to have a working area of 3x3 chunks, centered on the chunk that contains the miner. With the changes in this PR, that does not seem like it is still the case.

**Comment by ProTriforcer** [comment]
> the T1 miner is supposed to have a working area of 3x3 chunks, centered on the chunk that contains the miner. 

In chunk mode, yes. But it was actually 6x6, and the chunk mode didn't really work. I built a .jar with the changes and this is how it now mines with and without chunk mode.
image_2023-01-13_140112503

**Comment by ProTriforcer** [comment]
Were you confused by "Chunk Radius: 0"? Now that i'm looking at it, it could be worded better. Should it say "Area: 1x1 Chunks" instead? Should it tell you the area instead of block radius too?

**Comment by ALongStringOfNumbers** [comment]
The various tool tips are a bit confusing then, as they specify the working area as 1x1 chunks (both in the multiblock gui and tooltip), which is not correct. So could you fix those tool tips in this rework as well?

**Comment by ProTriforcer** [comment]
Oh, someone already changed those. Time to learn how to git

File: src/main/java/gregtech/api/capability/impl/miner/MinerLogic.java [hunk 1]
```diff
@@ -194,6 +194,7 @@ public void performMining() {
                 this.isDone = true;
                 this.wasActiveAndNeedsUpdate = true;
                 this.setActive(false);
+                this.setWorkingEnabled(false);
```
Comment (TechLord22) [inline src/main/java/gregtech/api/capability/impl/miner/MinerLogic.java:197]: Doesn't this mean that the miner will be in a paused state when it cannot find blocks to mine? Shouldn't it stay in an active but not mining state instead? If so, I think this line should be removed.

Comment (ProTriforcer) [reply src/main/java/gregtech/api/capability/impl/miner/MinerLogic.java:197]: It does, but i haven't noticed any issues with that. I guess i could move it to resetArea(), so it doesn't pause until you start changing the area.

File: src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityMiner.java [hunk 2]
```diff
@@ -130,20 +130,21 @@ protected ModularUI createUI(@Nonnull EntityPlayer entityPlayer) {
     }
 
     private void addDisplayText(@Nonnull List<ITextComponent> textList) {
+        int workingArea = getWorkingArea(minerLogic.getCurrentRadius());
         textList.add(new TextComponentTranslation("gregtech.machine.miner.startx", this.minerLogic.getX().get()));
         textList.add(new TextComponentTranslation("gregtech.machine.miner.starty", this.minerLogic.getY().get()));
         textList.add(new TextComponentTranslation("gregtech.machine.miner.startz", this.minerLogic.getZ().get()));
-        textList.add(new TextComponentTranslation("gregtech.machine.miner.radius", this.minerLogic.getCurrentRadius()));
+        textList.add(new TextComponentTranslation(I18n.format("gregtech.universal.tooltip.working_area", workingArea, workingArea)));
```
Comment (TechLord22) [inline src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityMiner.java:137]: This call to I18n.format() is unneeded, and will likely crash on a dedicated server.

File: src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityLargeMiner.java [hunk 3]
```diff
@@ -354,18 +358,30 @@ public boolean onScrewdriverClick(EntityPlayer playerIn, EnumHand hand, EnumFaci
...
+                else
+                    this.minerLogic.setCurrentRadius(currentRadius - CHUNK_LENGTH / 2);
+                int workingArea = getWorkingArea(minerLogic.getCurrentRadius());
+                playerIn.sendMessage(new TextComponentTranslation(I18n.format("gregtech.universal.tooltip.working_area", workingArea, workingArea)));
+            }
 
             this.minerLogic.resetArea();
 
-            playerIn.sendMessage(new TextComponentTranslation("gregtech.multiblock.large_miner.radius", this.minerLogic.getCurrentRadius()));
+
```
Comment (TechLord22) [inline src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityLargeMiner.java:382]: There's a lot of extra whitespace on this line which could be cleaned up.

File: src/main/java/gregtech/common/metatileentities/steam/SteamMiner.java [hunk 4]
```diff
@@ -135,22 +135,23 @@ protected ModularUI createUI(EntityPlayer entityPlayer) {
     }
 
     void addDisplayText(List<ITextComponent> textList) {
+        int workingArea = getWorkingArea(minerLogic.getCurrentRadius());
         textList.add(new TextComponentTranslation("gregtech.machine.miner.startx", this.minerLogic.getX().get()));
         textList.add(new TextComponentTranslation("gregtech.machine.miner.starty", this.minerLogic.getY().get()));
         textList.add(new TextComponentTranslation("gregtech.machine.miner.startz", this.minerLogic.getZ().get()));
-        textList.add(new TextComponentTranslation("gregtech.machine.miner.radius", this.minerLogic.getCurrentRadius()));
+        textList.add(new TextComponentTranslation(I18n.format("gregtech.universal.tooltip.working_area", workingArea, workingArea)));
```
Comment (TechLord22) [inline src/main/java/gregtech/common/metatileentities/steam/SteamMiner.java:142]: This call to I18n.format() is unneeded, and will likely crash on a dedicated server.

**Review by TechLord22 — CHANGES_REQUESTED** [review]

**Comment by ALongStringOfNumbers** [comment]
This has been incorporated in #1628

**Pull request closed without merging**